### PR TITLE
Fix PT023 not checking @pytest.marks in classes

### PR DESCRIPTION
**Pull request by and-semakin**
PT023 didn't check `@pytest.mark.foo` decorators in classes:

```python
class TestClass:
    @pytest.mark.foo()  # ok
    def test_something():
        pass

    @pytest.mark.foo  # also ok but should be an error
    def test_something_else():
        pass
```

Not it will recurse into classes and checks decorators on methods and nested classes.

I've also refactored tests for this rule because of DRY. Diff may look confusing. If it doesn't feel appropriate, please, let me know and I'll rewrite the tests.

### CODE REVIEW

File: tests/test_PT023_incorrect_mark_parentheses_style.py [hunk 1]
```diff
@@ -1,117 +1,155 @@
...
+                    def test_something():
+                        pass
+        """,
+        id='nested_class_method',
+    ),
+]
+
+
+@pytest.mark.parametrize('mark_parentheses', [True, False])
+def test_ok_with_parameters_no_matter_of_config(mark_parentheses: bool):
```
Comment (m-burst) [inline tests/test_PT023_incorrect_mark_parentheses_style.py:128]:
```suggestion
def test_ok_with_parameters_regardless_of_config(mark_parentheses: bool):
```

**Review by m-burst — APPROVED** [review]
Thanks for your contribution! Everything looks good, I've just made a minor suggestion in test naming

**Comment by m-burst** [comment]
Hi @and-semakin,
I've released v1.4.3 with this change.
Thanks again for your pull request!

**Comment by and-semakin** [comment]
Thanks! 🎉

**Pull request merged**